transactions/switchboard/setup_royalty_account.cdc

```diff
@@ -4,87 +4,68 @@ import "FlowToken"
+    let switchboardRef:  auth(FungibleTokenSwitchboard.Owner) &FungibleTokenSwitchboard.Switchboard
 
-    prepare(signer: AuthAccount) {
+    prepare(signer: auth(BorrowValue, IssueStorageCapabilityController, PublishCapability, SaveValue, UnpublishCapability) Account) {
 
-        // Check if the account already has a Flow Vault
-        if signer.borrow<&FlowToken.Vault>(from: /storage/flowTokenVault) == nil {
-            // If not, create a new Flow Vault resource and put it into storage
-            signer.save(<- FlowToken.createEmptyVault(vaultType: Type<@FlowToken.Vault>()), to: /storage/flowTokenVault)
-        }
-        // Check if the receiver capability is linked on the flow receiver path
-        if !signer.getCapability<&{FungibleToken.Receiver}>(/public/flowTokenReceiver)
-                                                                           .check() {
-            // if it's not, create a public capability to the flow vault exposing 
-            // the deposit function through the {FungibleToken.Receiver} interface
-            signer.unlink(/public/flowTokenReceiver)
-            signer.link<&FlowToken.Vault>(/public/flowTokenReceiver, 
-                                                     target: /storage/flowTokenVault)
-        }
-        self.flowTokenVaultCapability = signer.getCapability<&{FungibleToken.Receiver}>(/public/flowTokenReceiver)
+        self.flowTokenVaultCapability = signer.capabilities.get<&{FungibleToken.Receiver}>(/public/flowTokenReceiver)
+            ?? panic("Signer does not have a FlowToken receiver capability")
 
         // Check if the account already has a USDC Vault
-        if signer.borrow<&FiatToken.Vault>(from: FiatToken.VaultStoragePath) == nil {
+        if signer.storage.borrow<&FiatToken.Vault>(from: FiatToken.VaultStoragePath) == nil {
             // If not, create a new USDC Vault resource and put it into storage
-            signer.save(<- FiatToken.createEmptyVault(vaultType: Type<@FiatToken.Vault>()), 
+            signer.storage.save(<-FiatToken.createEmptyVault(vaultType: Type<@FiatToken.Vault>()), 
                                                       to: FiatToken.VaultStoragePath)
         }
         // Check if the receiver capability is linked on the USDC receiver path
-        if !signer.getCapability<&{FungibleToken.Receiver}>
-                                           (FiatToken.VaultReceiverPubPath).check() {
-            // if it's not, create a public capability to the USDC vault exposing 
-            // the deposit function through the {FungibleToken.Receiver} interface
-            signer.unlink(FiatToken.VaultReceiverPubPath)
-            signer.link<&FiatToken.Vault>(FiatToken.VaultReceiverPubPath, 
-                                                  target: FiatToken.VaultStoragePath)
+        if !signer.capabilities.get<&{FungibleToken.Receiver}>
+                                           (FiatToken.VaultReceiverPubPath)!.check() {
```

Comment:
Since `.get()` could return an optional here, I think it's better to check via
`if signer.capabilities.borrow<&{FungibleToken.Receiver}>(FiatToken.VaultReceiverPubPath) == nil`

Otherwise, I could see a case where the Vault is configured in the previous if block, `nil` is returned from `get()` since the Cap hasn't yet been published, and then forcing the optional would revert. Is that tracking or am I overthinking this?

Reply:
that is true. I haven't tested either of these transactions yet since we haven't finished the `FiatToken` upgrades yet, but I'll get to that once we're done.